t/03_collapse.t

```diff
@@ -7,27 +7,40 @@ use Data::NestedParams;
 
 cmp_deeply(
     collapse_nested_params({ a => undef }),
-    { a => undef },
+    [ a => undef ],
 );
 
 cmp_deeply(
     collapse_nested_params({ a => 3 }),
-    { a => 3 },
+    [ a => 3 ],
 );
 
 cmp_deeply(
     collapse_nested_params({ a => [qw(x y z)] }),
-    { 'a[]' => 'x', 'a[]' => 'y', 'a[]' => 'z'},
+    [ 'a[]' => 'x', 'a[]' => 'y', 'a[]' => 'z' ],
```

Comment:
Note: This test was broken because it was a hash ref with all 3 keys the same....  which means it was really a hash with only 1 element.